### PR TITLE
#181339534 Users should be able to make trip request

### PR DESCRIPTION
**Pull request by Tsa2341**
### what does this Pull Request do?

this Pull Request implements the CRUD operation of the trip

### description of tasks to be completed

 - user should be able to request a trip
 - user should be able to edit and delete trip requests with pending status
 - Manager should be able to view all trip requests he/she owns
 - implement pagination for getting trips

### how should this Pull Request be tested?

 - remottelly: **https://staging-bare-feature-18-lazhmv.herokuapp.com/api-docs**
 - locally: clone this repo and checkout **feature/#181339534-user-trip-request**  branch
 - add all environment variables and run npm run dev after running npm run dbmigrate:all to seed data
 - checkout this api to login
 
             http://localhost:5000/api/v1/users/login

- get the token and authenticate with a bearer and then hit these APIs to test the trips endpoints

           - **GET**   http://localhost:5000/api/v1/trips/?page=0&limit=2
           - **GET**   http://localhost:5000/api/v1/trips/manager/?page=0&limit=2
           - **POST**   http://localhost:5000/api/v1/trips
           - **PUT**   http://localhost:5000/api/v1/trips/1
           - **DELETE**   http://localhost:5000/api/v1/trips/1

### What are the relevant pivotal tracker stories?

 - [#181339534](https://www.pivotaltracker.com/story/show/181339534)

#### Screenshots (if appropriate)

Screenshot (33)
Screenshot (34)

### CODE REVIEW

**Review by Gabin-ishimwe — APPROVED** [review]

**Review by blandri — APPROVED** [review]

**Review by Code-250 — APPROVED** [review]

File: src/controllers/trip.controller.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        offset
+      );
+
+      return res.status(200).json({
+        message: 'Trip retrieved Successfully',
+        data: getPaginatedData(show, page, limit)
+      });
+    } catch (err) {
+      console.log(err);
+      return res.status(500).json({ message: 'internal server error', err });
```
Comment (ncutixavier) [inline src/controllers/trip.controller.js]: Provide appropriate error message

File: src/controllers/trip.controller.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          .status(404)
+          .json({ message: 'Manager with this Id not found' });
+      }
+
+      return res.status(200).json({
+        message: 'Trip retrieved Successfully',
+        data: getPaginatedData(managerFind, page, limit)
+      });
+    } catch (err) {
+      return res.status(500).json({ message: 'internal server error', err });
```
Comment (ncutixavier) [inline src/controllers/trip.controller.js]: Provide appropriate error message

File: src/controllers/trip.controller.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        req.body
+      );
+
+      return res.status(201).json({
+        message: 'Trip created successfully ',
+        data: tripCreated
+      });
+    } catch (err) {
+      console.log(err);
+      return res.status(500).json({ message: 'internal server error', err });
```
Comment (ncutixavier) [inline src/controllers/trip.controller.js]: Provide appropriate error message

File: src/services/trip.service.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        as: 'depart_location',
+        attributes: ['id', 'name', 'latitude', 'longitude', 'country']
+      },
+      {
+        model: Accommodation,
+        attributes: ['id', 'name', 'services', 'amenities', 'imagesId']
+      }
+    ];
+
+    if (userId === 1) {
```
Comment (ncutixavier) [inline src/services/trip.service.js]: Remove hardcoding

File: src/services/trip.service.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        as: 'depart_location',
+        attributes: ['id', 'name', 'latitude', 'longitude', 'country']
+      },
+      {
+        model: Accommodation,
+        attributes: ['id', 'name', 'services', 'amenities', 'imagesId']
+      }
+    ];
+
+    if (managerId === 1) {
```
Comment (ncutixavier) [inline src/services/trip.service.js]: Remove hardcoding

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**